integration-tests/src/test/java/org/apache/druid/tests/indexer/AbstractITBatchIndexTest.java

```diff
@@ -370,7 +370,12 @@ protected void submitTaskAndWait(
     if (segmentAvailabilityConfirmationPair.lhs != null && segmentAvailabilityConfirmationPair.lhs) {
       TaskReport reportRaw = indexer.getTaskReport(taskID).get("ingestionStatsAndErrors");
       IngestionStatsAndErrorsTaskReport report = (IngestionStatsAndErrorsTaskReport) reportRaw;
-      IngestionStatsAndErrors reportData = (IngestionStatsAndErrors) report.getPayload();
+      IngestionStatsAndErrors reportData = report.getPayload();
+
+      Assert.assertTrue(
+          reportData.getTaskContext() != null && !reportData.getTaskContext().isEmpty(),
+          "Report data does not contain task context. Ensure that TaskContextEnricher is correctly bound."
+      );
```

Comment:
Instead of just checking that the context is non empty - can we validate that certain fields we expect are there and contain the expected values?

Similar changes to existing integration tests for streaming, compaction and MSQ tasks please. I also noticed there are 2 `AbstractITBatchIndexTest` classes, this one, and another one in the `druid-it-cases` sub module. Please make a similar change there so that we get good integration test coverage that the task context is being included in the report for all the different test cases.

Reply:
+1